app/uk/gov/hmrc/cdsimportsddsfrontend/views/components/documentation_type.scala.html

```diff
@@ -21,10 +21,27 @@
 
 @(form: Form[Declaration])(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
 
-@input_text(form("documentationType.previousDocCategory" ), "declare.previousDocCategory" )
-@input_text(form("documentationType.previousDocType" ), "declare.previousDocType" )
-@input_text(form("documentationType.previousDocReference" ), "declare.previousDocReference" )
-@input_text(form("documentationType.previousDocGoodsItemId" ), "declare.previousDocGoodsItemId" )
+<table class="govuk-table">
+ <thead class="govuk-table__head">
+  <tr class="govuk-table__row">
+   <th colspan="4" class="govuk-table__header">@messages("declare.previousDocuments")</th>
+  </tr>
+ </thead>
+ <tbody class="govuk-table__body">
+  @for((doc, count) <- form.value.getOrElse(Declaration()).documentationType.previousDocument.zipWithIndex) {
```

Comment:
This couples the number of fields displayed on the form with the number of defaults in the domain class. Probably safer to use a range 0 to 5 here